### PR TITLE
RUMM-2745 Decouple Webview Tracking

### PR DESCRIPTION
**Pull request by maxep**
### What and why?

The web-view tracking no longer depends on the `DatadogV1Context` and will send raw logs and events through the message-bus.

The `DatadogV1Context` can now be removed as well as its dependencies:
- `CarrierInfoProvider`
- `NetworConnectionInfoProvider`
- `AppVersionProvider`
- `DateCorrector`

### How?

The bridge used to inject native context in Browser events, this responsibility moves to the Feature message receiver. The `WebEventBridge` now send raw Browser events on the bus, the Log and RUM receiver will be able to inject current context (e.g ddTags, or RUM context) before writing it to `core`.

### Review checklist
- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference
- [ ] Add CHANGELOG entry for user facing changes

### Custom CI job configuration (optional)
- [x] Run unit tests
- [x] Run integration tests
- [ ] Run smoke tests

### CODE REVIEW

File: Sources/Datadog/RUM/Instrumentation/RUMCommandSubscriber.swift [hunk 1]
```diff
@@ -27,3 +27,10 @@ internal protocol RUMCommandPublisher: AnyObject {
     /// - Parameter subscriber: The RUM command subscriber.
     func publish(to subscriber: RUMCommandSubscriber)
 }
+
+internal final class GlobalRUMCommandSubscriber: RUMCommandSubscriber {
+    func process(command: RUMCommand) {
+        let subscriber = Global.rum as? RUMCommandSubscriber
+        subscriber?.process(command: command)
+    }
+}
```
Comment (maxep) [inline Sources/Datadog/RUM/Instrumentation/RUMCommandSubscriber.swift:36]: I'm not a fan of this, but it helps for injecting a `RUMCommandSubscriber` instance for testing. Ultimately, The `Global.rum` and this class will be removed.

File: Sources/Datadog/RUM/Integrations/WebViewEventReceiver.swift [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-2020 Datadog, Inc.
+ */
+
+import Foundation
+
+internal typealias JSON = [String: Any]
+
+/// Receiver to consume a RUM event coming from Browser SDK.
+internal final class WebViewEventReceiver {
```
Comment (maxep) [inline Sources/Datadog/RUM/Integrations/WebViewEventReceiver.swift:12]: This logic is migrated from `WebRUMEventConsumer`.

File: Sources/Datadog/RUM/Instrumentation/RUMCommandSubscriber.swift (hunk 1 above)
Comment (maciejburda) [reply Sources/Datadog/RUM/Instrumentation/RUMCommandSubscriber.swift:36]: Maybe worth documenting the usage and plan in the header comment?

**Review by maciejburda — APPROVED** [review]
LGTM

File: Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift [hunk 3]
```diff
@@ -161,17 +161,23 @@ class LoggingMessageReceiverTests: XCTestCase {
         )
 
         // When
-        let sent = Event(test: .mockRandom())
```
Comment (ncreated) [inline Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift:164]: Cleanup/ we can now remove the `Event` declaration above:
```swift
struct Event: Encodable {
   let test: String
}
```

File: Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift [hunk 4]
```diff
@@ -196,4 +202,96 @@ class LoggingMessageReceiverTests: XCTestCase {
...
+            message: .custom(
+                key: LoggingMessageKeys.browserLog,
+                baggage: .init(webLogEvent)
+            )
+        )
+
+        let data = try JSONEncoder().encode(core.events.first as? AnyEncodable)
+        let writtenJSON = try XCTUnwrap(try JSONSerialization.jsonObject(with: data, options: []) as? JSON)
+
+        AssertDictionariesEqual(writtenJSON, expectedWebLogEvent)
```
Comment (ncreated) [inline Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift:258]: nit/ Wouldn't this be more convenient?
```suggestion
        let received = try XCTUnwrap(core.events.first, "It should send event")
        try AssertEncodedRepresentationsEqual(received, AnyEncodable(expectedWebLogEvent))
```

**Review by ncreated — APPROVED** [review]
Well done 🎯. Happy to see remaining V1 context gone 💪 and the use of V2 things for further separation of Features.

Everything makes total sense 👍 and `AnyCodable` pays-off well. I left few minor comments, but approve it anyway.

File: Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift (hunk 4 above)
Comment (maxep) [reply Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift:258]: Yes indeed 👍

File: Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift (hunk 3 above)
Comment (maxep) [reply Tests/DatadogTests/Datadog/Logging/LoggingMessageReceiverTests.swift:164]: Ah good catch!! 👀

**Pull request merged**